guidelines/enhancement_template.md

```diff
@@ -272,6 +272,40 @@ to implement the design.  For instance,
  > 1. This requires exposing previously private resources which contain sensitive
   information.  Can we do this?
 
+## Security
+
+This section should cover the security impact of the change, if any. For example,
+does it introduce new privileges or capabilities? Does it expose new risks?
+
+### Credentials and RBAC
+
+What credentials will the new component be using if it needs to talk to either the control plane or other components? What RBAC needs to be applied?
+
+Consider the following:
+- Which service accounts will be affected by the change, what permissions will the feature code need?
+- Will end users need to make any security considerations when enabling this feature?
+- What is the scope of the required RBAC?
+  - Are cluster wider permissions required?
+  - Should RBAC be limited to only the namespace(s) where the feature is enabled?
+  - Should RBAC be limited to just the node(s) where the feature is enabled?
+  - Are we adding new permissions or escalation paths from Nodes to the Cluster level?
+
+Principles to consider:
+- The permissions you grant should be well scoped and follow the principle of least privilege. Do not use wildcard permissions.
+- If the feature operates at the host level, the host root user should not be able to escalate to the cluster level. The feature may only modify resources directly related to the node it operates on.
+
+Consider the above principles as mandatory. These should be considered as part of your E2E test plan. How will you ensure that the principles are upheld?
+
+### Security Context
+
+What security context will the new component run with? What is the impact of the security context on the security posture of the cluster?
```

Comment:
We added an e2e test to ensure all components specify which SCC they want.  Just an example of something similar to the RBAC bits.